src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -573,9 +593,33 @@ private void ProcessEvents()
                                 watcher.NotifyFileSystemEventArgs(WatcherChangeTypes.Changed, expandedName);
                                 break;
                             case Interop.libc.NotifyEvents.IN_MOVED_FROM:
-                                previousEventName = expandedName;
-                                previousEventParent = isDir ? associatedDirectoryEntry : null;
-                                previousEventCookie = nextEvent.cookie;
+                                // We need to check if this MOVED_FROM event is standalone - meaning the item was moved out
+                                // of scope. We do this by checking if we are at the end of our buffer (meaning no more events) 
+                                // and if there is data to be read by polling the fd. If there aren't any more events, fire the
+                                // deleted event; if there are more events, handle it via next pass.
```

Comment:
Maybe add an additional comment about this potentially missing the case where the FROM event arrived and a TO event was coming but hasn't arrived yet. In that case we'll end up raising a delete + create instead of a rename, but we already do that in some other cases (e.g. if there are intervening events between the FROM and TO), and this coded behavior is least worst of the options, as it means we'll always raise an event for a FROM even if it's the last one for a while.